Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py

```diff
@@ -8457,9 +8460,117 @@ def test_send_events_to_xsiam_error_handling(self, mocker, requests_mock, error_
         ):
             send_events_to_xsiam(events=events, vendor='some vendor', product='some product')
 
+        # make sure the request was sent only once and retry mechanism was not triggered
+        assert request_mocker.call_count == 1
+
         error_log_mocker.assert_called_with(
             expected_request_and_response_info.format(status_code=str(status_code), error_received=expected_error_msg))
 
+    @pytest.mark.parametrize(
+        'mocked_responses, expected_request_call_count, expected_error_log_count, should_succeed', [
+            (
+                [
+                    (429, None), (429, None), (429, None)
+                ],
+                3,
+                3,
+                False
+            ),
+            (
+                [
+                    (401, None), (401, None)
```

Comment:
I think we don't use the second item here, why do we need it?
```suggestion
                    (401, None)
```